src/cursor/cur_version.c

```diff
@@ -0,0 +1,432 @@
+              tw.stop_ts, tw.durable_stop_ts, hs_upd_type, 0, 0, WT_VERSION_HISTORY_STORE);
+
+            /* Reconstruct the history store value if needed. */
+            if (hs_upd_type == WT_UPDATE_MODIFY) {
+                WT_ERR(__wt_modify_apply_item(
+                  session, table_cursor->value_format, version_cursor->hs_base_upd, hs_value.data));
+                cursor->value.data = version_cursor->hs_base_upd->data;
+                cursor->value.size = version_cursor->hs_base_upd->size;
+            } else {
+                WT_ASSERT(session, hs_upd_type == WT_UPDATE_STANDARD);
+                cursor->value.data = hs_value.data;
+                cursor->value.size = hs_value.size;
+                version_cursor->hs_base_upd->data = hs_value.data;
+                version_cursor->hs_base_upd->size = hs_value.size;
+            }
+            F_SET(cursor, WT_CURSTD_VALUE_EXT);
+            upd_found = true;
+        } else
+            F_SET(version_cursor, WT_VERSION_CUR_HS_EXAUSTED);
+    }
+
+done:
+    API_END_RET(session, ret);
+
+    if (0) {
+err:
+        WT_TRET(cursor->reset(cursor));
+    }
+}
+
+/*
+ * __curversion_reset --
+ *     WT_CURSOR::reset for version cursors.
+ */
+static int
+__curversion_reset(WT_CURSOR *cursor)
+{
+    WT_CURSOR *hs_cursor, *table_cursor;
+    WT_CURSOR_VERSION *version_cursor;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    version_cursor = (WT_CURSOR_VERSION *)cursor;
+    hs_cursor = version_cursor->hs_cursor;
+    table_cursor = version_cursor->table_cursor;
+    CURSOR_API_CALL(cursor, session, reset, NULL);
+
+    if (table_cursor != NULL)
+        WT_TRET(table_cursor->reset(table_cursor));
+    if (hs_cursor != NULL)
+        WT_TRET(hs_cursor->reset(hs_cursor));
+    version_cursor->next_upd = NULL;
+    version_cursor->flags = 0;
+    F_CLR(cursor, WT_CURSTD_KEY_SET);
+    F_CLR(cursor, WT_CURSTD_VALUE_SET);
+
+err:
+    __wt_scr_free(session, &(version_cursor->hs_base_upd));
+    API_END_RET(session, ret);
+}
+
+/*
+ * __curversion_search --
+ *     WT_CURSOR->search method for version cursors.
+ */
+static int
+__curversion_search(WT_CURSOR *cursor)
+{
+    WT_CURSOR *table_cursor;
+    WT_CURSOR_BTREE *cbt;
+    WT_CURSOR_VERSION *version_cursor;
+    WT_DECL_RET;
+    WT_INSERT *ins;
+    WT_PAGE *page;
+    WT_ROW *rip;
+    WT_SESSION_IMPL *session;
+    WT_UPDATE *upd;
+    bool key_only;
+
+    version_cursor = (WT_CURSOR_VERSION *)cursor;
+    table_cursor = version_cursor->table_cursor;
+    key_only = F_ISSET(cursor, WT_CURSTD_KEY_ONLY);
+
+    /*
+     * For now, we assume that we are using simple cursors only.
+     */
+    cbt = (WT_CURSOR_BTREE *)table_cursor;
+    CURSOR_API_CALL(cursor, session, search, CUR2BT(cbt));
+    WT_ERR(__cursor_checkkey(table_cursor));
+
+    /* Do a search and position on they key if it is found */
+    F_SET(cursor, WT_CURSTD_KEY_ONLY);
+    WT_ERR(__wt_btcur_search(cbt));
+    if (!F_ISSET(cbt, WT_CURSTD_KEY_SET))
```

Comment:
Should check WT_CURSTD_KEY_INT

Reply:
Actually no need to check this as __wt_btcur_search should return not found already.
You can turn this check to an assert.